packages/massa-web3/src/experimental/smartContract.ts

```diff
@@ -225,7 +230,7 @@ export class SmartContract {
       func,
       parameter,
       targetAddress: this.contractAddress,
-      callerAddress: opts.callerAddress || this.contractAddress,
+      callerAddress: opts.callerAddress ?? this.account.address.toString(),
```

Comment:
Why callerAddress is not an address type? Why setting the type in the name of that field?

Reply:
you're right